### PR TITLE
Configuration wizard

### PR DESCRIPTION
**Pull request by sirjla**
# Resumen
Se implemento landing page y configuration wizard
# Detalles
- Se implemento landing page para autentificacion con trello
- Se implemento un configuration wizard para un solo board

### CODE REVIEW

File: src/components/BoardWizard.vue [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+              <label :for="`prod_${list.id}`" class="checkbox" v-on:click="generalListChanged"></label>
+              <label :for="`prod_${list.id}`" class="wizard--text-list">{{ list.name }}</label>
+            </div>
+          </div>
+        </div>
+        <div>
+          <p class="wizard--text">Archived</p>
+          <div class="wizard--container wizard--container-list wizard--container-list-special">
+            <div class="checkbox-container" v-for="list in selectedListsObjects" :key="list.id">
+              <input type="checkbox" :id="`arc_${list.id}`" style="display: none;" :value="list.id" v-model="archivedLists">
```
Comment (monkeyci) [inline src/components/BoardWizard.vue]: Line 78 exceeds the maximum line length of 120  max-len

File: src/components/BoardWizard.vue [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+              <label :for="`end_${list.id}`" class="checkbox" v-on:click="generalListChanged"></label>
+              <label :for="`end_${list.id}`" class="wizard--text-list">{{ list.name }}</label>
+            </div>
+          </div>
+        </div>
+        <div>
+          <p class="wizard--text">Production</p>
+          <div class="wizard--container wizard--container-list wizard--container-list-special">
+            <div class="checkbox-container" v-for="list in selectedListsObjects" :key="list.id">
+              <input type="radio" :id="`prod_${list.id}`" style="display: none;" :value="list.id" v-model="productionList">
```
Comment (monkeyci) [inline src/components/BoardWizard.vue]: Line 68 exceeds the maximum line length of 120  max-len

File: src/components/BoardWizard.vue [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+  <div class="wizard">
+    <div class="wizard--progress-bar"></div>
+    <div class="wizard--progress-bar wizard--progress-bar--step-1" id="progress_bar"></div>
+    <div id="main_container" class="wizard--container">
+      <div v-if="stage === 0" class="wizard--container wizard--container-inner">
+        <p class="wizard--title">Choose a Board</p>
+        <p class="wizard--text">Choose a Trello Board and press Next.</p>
+        <p class="wizard--text wizard--text-board-container">Boards</p>
+        <div style="width: 100%;"></div>
+        <BoardBox v-for="board in boards" :key="board.id" :id="board.id" :board="board" v-on:click.native="selectBoard"/>
```
Comment (monkeyci) [inline src/components/BoardWizard.vue]: Line 11 exceeds the maximum line length of 120  max-len

File: src/assets/style.css [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+.checkbox-container input[type=radio] {
+  display: none;
+}
+input[type=radio]:checked + .checkbox {
+  border-radius: 10px;
+  background-color: #5CEAA5;
+}
+.wizard--container-wide {
+  width: 80%;
+}
```
Comment (monkeyci) [inline src/assets/style.css]: Unexpected missing end-of-source newline                    no-missing-end-of-source-newline

File: src/assets/style.css [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+  background-color: #5CEAA5;
+}
+.checkbox-container input[type=radio] {
+  display: none;
+}
+input[type=radio]:checked + .checkbox {
+  border-radius: 10px;
+  background-color: #5CEAA5;
+}
+.wizard--container-wide {
```
Comment (monkeyci) [inline src/assets/style.css]: Expected empty line before rule                             rule-empty-line-before

File: src/assets/style.css [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+input[type=checkbox]:checked + .checkbox {
+  border-radius: 10px;
+  background-color: #5CEAA5;
+}
+.checkbox-container input[type=radio] {
+  display: none;
+}
+input[type=radio]:checked + .checkbox {
+  border-radius: 10px;
+  background-color: #5CEAA5;
```
Comment (monkeyci) [inline src/assets/style.css]: Expected "#5CEAA5" to be "#5ceaa5"                          color-hex-case

File: src/assets/style.css [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+  display: none;
+}
+input[type=checkbox]:checked + .checkbox {
+  border-radius: 10px;
+  background-color: #5CEAA5;
+}
+.checkbox-container input[type=radio] {
+  display: none;
+}
+input[type=radio]:checked + .checkbox {
```
Comment (monkeyci) [inline src/assets/style.css]: Expected empty line before rule                             rule-empty-line-before<br>Unexpected qualifying type selector                         selector-no-qualifying-type

File: src/assets/style.css [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+  z-index: 1;
+}
+.checkbox-container input[type=checkbox] {
+  display: none;
+}
+input[type=checkbox]:checked + .checkbox {
+  border-radius: 10px;
+  background-color: #5CEAA5;
+}
+.checkbox-container input[type=radio] {
```
Comment (monkeyci) [inline src/assets/style.css]: Expected empty line before rule                             rule-empty-line-before<br>Unexpected qualifying type selector                         selector-no-qualifying-type

File: src/assets/style.css [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+  display: inline-block;
+  background-color: #F0F0F0;
+  z-index: 1;
+}
+.checkbox-container input[type=checkbox] {
+  display: none;
+}
+input[type=checkbox]:checked + .checkbox {
+  border-radius: 10px;
+  background-color: #5CEAA5;
```
Comment (monkeyci) [inline src/assets/style.css]: Expected "#5CEAA5" to be "#5ceaa5"                          color-hex-case

File: src/assets/style.css [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+  height: 20px;
+  width: 20px;
+  display: inline-block;
+  background-color: #F0F0F0;
+  z-index: 1;
+}
+.checkbox-container input[type=checkbox] {
+  display: none;
+}
+input[type=checkbox]:checked + .checkbox {
```
Comment (monkeyci) [inline src/assets/style.css]: Expected empty line before rule                             rule-empty-line-before<br>Unexpected qualifying type selector                         selector-no-qualifying-type

File: src/components/BoardWizard.vue [hunk 11]
```diff
@@ -0,0 +1,272 @@
...
+  updated() {
+    if (this.stage === 1 && this.toLoad === true) {
+      console.log('IN');
+      this.selectedLists.forEach((list) => {
+        const elem = document.getElementById(`selec_${list}`);
+        console.log(elem);
+        if (elem != null) elem.parentElement.classList.toggle('checkbox-container-selected');
+      });
+      this.toLoad = false;
+    } else if (this.stage === 2 && this.toLoad) {
```
Comment (monkeyci) [inline src/components/BoardWizard.vue]: No magic number: 2                   no-magic-numbers

File: src/components/BoardWizard.vue [hunk 12]
```diff
@@ -0,0 +1,272 @@
...
+        .classList.toggle('wizard--board-selected');
+    }
+  },
+  updated() {
+    if (this.stage === 1 && this.toLoad === true) {
+      console.log('IN');
+      this.selectedLists.forEach((list) => {
+        const elem = document.getElementById(`selec_${list}`);
+        console.log(elem);
+        if (elem != null) elem.parentElement.classList.toggle('checkbox-container-selected');
```
Comment (monkeyci) [inline src/components/BoardWizard.vue]: Expected '!==' and instead saw '!='  eqeqeq

File: src/assets/style.css [hunk 13]
```diff
@@ -0,0 +1,278 @@
...
+
+.checkbox {
+  height: 20px;
+  width: 20px;
+  display: inline-block;
+  background-color: #F0F0F0;
+  z-index: 1;
+}
+
+.checkbox-container input[type=checkbox] {
```
Comment (monkeyci) [inline src/assets/style.css]: Unexpected qualifying type selector                         selector-no-qualifying-type

File: src/assets/style.css [hunk 14]
```diff
@@ -0,0 +1,278 @@
...
+
+.checkbox-container-selected {
+  background-color: rgba(92, 234, 165, 0.08);
+}
+
+.checkbox {
+  height: 20px;
+  width: 20px;
+  display: inline-block;
+  background-color: #F0F0F0;
```
Comment (monkeyci) [inline src/assets/style.css]: Expected "#F0F0F0" to be "#f0f0f0"                          color-hex-case

File: src/assets/style.css [hunk 15]
```diff
@@ -0,0 +1,278 @@
...
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  align-items: center;
+  margin: 8px;
+  z-index: 0;
+}
+
+.checkbox-container-selected {
+  background-color: rgba(92, 234, 165, 0.08);
```
Comment (monkeyci) [inline src/assets/style.css]: Unexpected leading zero                                     number-leading-zero

File: src/assets/style.css [hunk 16]
```diff
@@ -0,0 +1,278 @@
...
+
+.wizard--container-list-special {
+  width: 95%;
+}
+
+.checkbox-container {
+  width: max-content;
+  height: 42px;
+  border-radius: 21px;
+  background-color: #ffffff;
```
Comment (monkeyci) [inline src/assets/style.css]: Expected "#ffffff" to be "#fff"                             color-hex-length

File: src/assets/style.css [hunk 17]
```diff
@@ -0,0 +1,278 @@
...
+  border: none;
+}
+
+.wizard--container-list {
+  margin-top: 10px;
+  height: 484px;
+  width: 60%;
+  border: 1px solid rgba(255,214,234,0.57);
+  border-radius: 8px;
+  box-shadow: 0 9px 11px 0 rgba(85,57,255,0.06);
```
Comment (monkeyci) [inline src/assets/style.css]: Expected single space after "," in a single-line function   function-comma-space-after<br>Unexpected leading zero                                     number-leading-zero

File: src/assets/style.css [hunk 18]
```diff
@@ -0,0 +1,278 @@
...
+.button-save {
+  background-color: #fcea10;
+  border: none;
+}
+
+.wizard--container-list {
+  margin-top: 10px;
+  height: 484px;
+  width: 60%;
+  border: 1px solid rgba(255,214,234,0.57);
```
Comment (monkeyci) [inline src/assets/style.css]: Expected single space after "," in a single-line function   function-comma-space-after<br>Unexpected leading zero                                     number-leading-zero

File: src/assets/style.css [hunk 19]
```diff
@@ -0,0 +1,278 @@
...
+  font-family: Rubik;
+  font-size: 20px;
+  font-weight: 500;
+  letter-spacing: 0.4px;
+  line-height: 24px;
+}
+
+.button-save {
+  background-color: #fcea10;
+  border: none;
```
Comment (monkeyci) [inline src/assets/style.css]: Unexpected value "none" for property "border"               declaration-property-value-blacklist

File: src/assets/style.css [hunk 20]
```diff
@@ -0,0 +1,278 @@
...
+  height: 56px;
+  width: 200px;
+  border: 1px solid #211952;
+  background-color: #ffffff;
+  margin: 5px;
+  color: #211952;
+  font-family: Rubik;
+  font-size: 20px;
+  font-weight: 500;
+  letter-spacing: 0.4px;
```
Comment (monkeyci) [inline src/assets/style.css]: Unexpected leading zero                                     number-leading-zero

File: src/components/LandingPage.vue [hunk 21]
```diff
@@ -0,0 +1,25 @@
+<template>
+  <div class="landing">
+    <img src="../assets/Platanus.png" class="logo-box">
+    <div class="trello" v-on:click="trelloAuthorize"><img src="../assets/trello-logo.png" class="trello-logo"></div>
```
Comment (llekn) [inline src/components/LandingPage.vue]: No se ve!

File: src/assets/style.css [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+  height: 768px;
+  width: 1366px;
+  overflow: hidden;
+  position: relative;
+}
+.landing-rectangle {
+	height: 178px;
+	width: 1196px;
+	background-color: #FFD6EA;
+  position: absolute;
```
Comment (llekn) [inline src/assets/style.css]: ¿¿Nooooo por qué usar posicionamiento absoluto?? 😢

File: src/assets/style.css [hunk 23]
```diff
@@ -0,0 +1,65 @@
+.landing {
+  height: 768px;
```
Comment (llekn) [inline src/assets/style.css]: Noooo... por qué usar tamaños fijos en px?

File: src/components/LandingPage.vue [hunk 24]
```diff
@@ -2,7 +2,7 @@
   <div class="landing">
     <img src="../assets/Platanus.png" class="logo-box">
     <div class="trello" v-on:click="trelloAuthorize"><img src="../assets/trello-logo.png" class="trello-logo"></div>
-    <p class="landing-text">Beef ribs landjaeger meatloaf burgdoggen, spare ribs shank doner pastrami kevin bacon.</p>
+    <p align="left" class="landing-text">Denme mis malditas métricas de proyecto</p>
```
Comment (llekn) [inline src/components/LandingPage.vue]: Hay que manejar los estilos de presentación en css!

**Comment by sirjla** [comment]
[Implementar Wizard de configuración inicial de board](https://trello.com/c/lLyxSb7B/85-implementar-wizard-de-configuraci%C3%B3n-inicial-de-board)

File: src/components/BoardWizard.vue [hunk 25]
```diff
@@ -66,6 +162,44 @@ export default {
...
+        () => {
+          onRequestError(self.getLists, [boardId]);
+        }
+      );
+    },
+    generalListChanged(event) {
+      event.target.parentElement.classList.toggle('checkbox-container-selected');
+    },
+    saveSpecificLists() {
+      console.log('WIIIIII');
```
Comment (llekn) [inline src/components/BoardWizard.vue]: ?

File: src/components/BoardWizard.vue [hunk 26]
```diff
@@ -0,0 +1,333 @@
...
+      progressBar.classList.remove('wizard--progress-bar--step-1');
+      progressBar.classList.add('wizard--progress-bar--step-2');
+
+      this.stage++;
+      this.toLoad = true;
+    },
+    leaveWizard() {
+      this.$emit('leaveWizard', true);
+    },
+    stepBack(){
```
Comment (monkeyci) [inline src/components/BoardWizard.vue]: Method 'stepBack' has too many statements (13). Maximum allowed is 10  max-statements<br>Missing space before opening brace                                     space-before-blocks

File: src/components/BoardWizard.vue [hunk 27]
```diff
@@ -0,0 +1,333 @@
...
+  },
+  methods: {
+    selectableLists(usedLists) {
+      const uLists = (usedLists === null || usedLists === undefined) ? [] : usedLists;
+      return this.allLists.filter((list) => this.selectedLists.includes(list.id))
+        .filter((list) => this.backlogList !== list.id)
+        .filter((list) => !this.wipLists.includes(list.id))
+        .filter((list) => this.endList !== list.id)
+        .filter((list) => this.productionList !== list.id)
+        .concat(this.allLists.filter((aList) => uLists.includes(aList.id)))
```
Comment (monkeyci) [inline src/components/BoardWizard.vue]: Missing semicolon                                                      semi

File: src/components/BoardWizard.vue [hunk 28]
```diff
@@ -0,0 +1,333 @@
...
+      if (elem !== null) elem.parentElement.classList.toggle('checkbox-container-selected');
+      elem = document.getElementById(`prod_${this.productionList}`);
+      if (elem !== null) elem.parentElement.classList.toggle('checkbox-container-selected');
+      this.toLoad = false;
+    }
+  },
+  methods: {
+    selectableLists(usedLists) {
+      const uLists = (usedLists === null || usedLists === undefined) ? [] : usedLists;
+      return this.allLists.filter((list) => this.selectedLists.includes(list.id))
```
Comment (monkeyci) [inline src/components/BoardWizard.vue]: Expected newline before return statement                               newline-before-return

File: src/components/BoardWizard.vue [hunk 29]
```diff
@@ -0,0 +1,333 @@
...
+      });
+      this.toLoad = false;
+    } else if (this.stage === secondStageNumber && this.toLoad) {
+      this.wipLists.forEach((list) => {
+        elem = document.getElementById(`wip_${list}`);
+        if (elem !== null) elem.parentElement.classList.toggle('checkbox-container-selected');
+      });
+      this.archivedLists.forEach((list) => {
+        elem = document.getElementById(`arc_${list}`);
+        if (elem !== null) elem.parentElement.classList.toggle('checkbox-container-selected')
```
Comment (monkeyci) [inline src/components/BoardWizard.vue]: Missing semicolon                                                      semi

File: src/components/BoardWizard.vue [hunk 30]
```diff
@@ -0,0 +1,333 @@
...
+      this.selectedBoard = this.selectedBoard[0]
+      document.getElementById(this.selectedBoard)
+        .classList.toggle('wizard--board-selected');
+      document.getElementById('save_button').classList.remove('button-disabled');
+      document.getElementById('save_button').disabled = false;
+    }
+  },
+  updated() {
+    let elem;
+    if (this.stage === 0 && this.selectedBoard != null) {
```
Comment (monkeyci) [inline src/components/BoardWizard.vue]: Expected '!==' and instead saw '!='                                    eqeqeq

File: src/components/BoardWizard.vue [hunk 31]
```diff
@@ -0,0 +1,333 @@
...
+    this.selectedBoard = get('boards', null);
+    if (this.selectedBoard !== null) {
+      this.selectedBoard = this.selectedBoard[0]
+      document.getElementById(this.selectedBoard)
+        .classList.toggle('wizard--board-selected');
+      document.getElementById('save_button').classList.remove('button-disabled');
+      document.getElementById('save_button').disabled = false;
+    }
+  },
+  updated() {
```
Comment (monkeyci) [inline src/components/BoardWizard.vue]: Method 'updated' has too many statements (14). Maximum allowed is 10   max-statements

File: src/components/BoardWizard.vue [hunk 32]
```diff
@@ -0,0 +1,333 @@
...
+      return this.boards.find((board) => board.id === this.selectedBoard);
+    },
+    selectedListsObjects() {
+      return this.allLists.filter((list) => this.selectedLists.includes(list.id));
+    },
+  },
+  mounted() {
+    this.selectedBoard = get('boards', null);
+    if (this.selectedBoard !== null) {
+      this.selectedBoard = this.selectedBoard[0]
```
Comment (monkeyci) [inline src/components/BoardWizard.vue]: Missing semicolon                                                      semi

File: src/assets/style.css [hunk 33]
```diff
@@ -0,0 +1,301 @@
...
+  height: 484px;
+  width: 60%;
+  border: 1px solid rgba(255,214,234,0.57);
+  border-radius: 8px;
+  box-shadow: 0 9px 11px 0 rgba(85,57,255,0.06);
+  display: block;
+  overflow: auto;
+}
+
+.wizard--container-list-special {
```
Comment (monkeyci) [inline src/assets/style.css]: Expected empty line before rule                             rule-empty-line-before

File: src/assets/style.css [hunk 34]
```diff
@@ -0,0 +1,301 @@
...
+}
+
+.wizard--container-list {
+  margin-top: 10px;
+  height: 484px;
+  width: 60%;
+  border: 1px solid rgba(255,214,234,0.57);
+  border-radius: 8px;
+  box-shadow: 0 9px 11px 0 rgba(85,57,255,0.06);
+  display: block;
```
Comment (monkeyci) [inline src/assets/style.css]: Expected empty line before rule                             rule-empty-line-before<br>Unexpected qualifying type selector                         selector-no-qualifying-type

File: src/assets/style.css [hunk 35]
```diff
@@ -0,0 +1,301 @@
...
+.button-save {
+  background-color: #fcea10;
+  border: none;
+}
+
+.wizard--container-list {
+  margin-top: 10px;
+  height: 484px;
+  width: 60%;
+  border: 1px solid rgba(255,214,234,0.57);
```
Comment (monkeyci) [inline src/assets/style.css]: Expected empty line before rule                             rule-empty-line-before<br>Unexpected qualifying type selector                         selector-no-qualifying-type

File: src/assets/style.css [hunk 36]
```diff
@@ -0,0 +1,301 @@
...
+  background-color: #eaeaea;
+  opacity: 0.5;
+}
+
+.button-save {
+  background-color: #fcea10;
+  border: none;
+}
+
+.wizard--container-list {
```
Comment (monkeyci) [inline src/assets/style.css]: Expected empty line before rule                             rule-empty-line-before<br>Unexpected qualifying type selector                         selector-no-qualifying-type

**Pull request merged**